service/history/interfaces/mutable_state.go

```diff
@@ -355,7 +355,9 @@ type (
 		// activities.
 		// If there is a pending workflow task that is not started yet, it'll be rescheduled after
 		// transition start.
-		StartDeploymentTransition(deployment *deploymentpb.Deployment) error
+		StartDeploymentTransition(deployment *deploymentpb.Deployment, revisionNumber int64) error
+		GetRevisionNumber() int64
```

Comment:
Better to call these `Get/SetVersioningRevisionNumber` because in mutable state there are other concepts close to this that can make ppl confused.